interfaces/builtin/kubernetes_support.go

```diff
@@ -35,65 +47,226 @@ const kubernetesSupportBaseDeclarationSlots = `
     deny-auto-connection: true
 `
 
-const kubernetesSupportConnectedPlugAppArmor = `
-# Description: can use kubernetes to control Docker containers. This interface
-# is restricted because it gives wide ranging access to the host and other
-# processes.
+const kubernetesSupportConnectedPlugAppArmorCommon = `
+# Common rules for running as a kubernetes node
 
-# what is this for?
-#include <abstractions/dbus-strict>
-
-# Allow reading all information regarding cgroups for all processes
+# reading cgroups
 capability sys_resource,
-@{PROC}/@{pid}/cgroup r,
 /sys/fs/cgroup/{,**} r,
 
-# Allow adjusting the OOM score for Docker containers. Note, this allows
-# adjusting for all processes, not just containers.
+# Allow adjusting the OOM score for containers. Note, this allows adjusting for
+# all processes, not just containers.
 @{PROC}/@{pid}/oom_score_adj rw,
-/sys/kernel/mm/hugepages/ r,
+@{PROC}/sys/vm/overcommit_memory rw,
+/sys/kernel/mm/hugepages/{,**} r,
+
+capability dac_override,
+
+/usr/bin/systemd-run Cxr -> systemd_run,
+profile systemd_run (attach_disconnected,mediate_deleted) {
+  # Common rules for kubernetes use of systemd_run
+  #include <abstractions/base>
+
+  /usr/bin/systemd-run r,
+  owner @{PROC}/@{pid}/stat r,
+  owner @{PROC}/@{pid}/environ r,
+  @{PROC}/cmdline r,
+
+  # setsockopt()
+  capability net_admin,
+
+  /run/systemd/private rw,
+  /bin/true ixr,
+  ptrace trace peer=unconfined,
+###KUBERNETES_SUPPORT_SYSTEMD_RUN###
+}
+`
+
+const kubernetesSupportConnectedPlugAppArmorKubelet = `
+# Allow running as the kubelet service
+
+# Ideally this would be snap-specific
+/run/dockershim.sock rw,
+
+# allow managing pods' cgroups
+/sys/fs/cgroup/*/kubepods/{,**} rw,
 
-@{PROC}/sys/kernel/panic rw,
-@{PROC}/sys/kernel/panic_on_oops rw,
+# Allow tracing our own processes. Note, this allows seccomp sandbox escape on
+# kernels < 4.8
+capability sys_ptrace,
+ptrace (trace) peer=snap.@{SNAP_NAME}.*,
+
+# Allow ptracing other processes (as part of ps-style process lookups). Note,
+# the peer needs a corresponding tracedby rule. As a special case, disallow
+# ptracing unconfined.
+ptrace (trace),
+deny ptrace (trace) peer=unconfined,
+
+@{PROC}/[0-9]*/attr/ r,
+@{PROC}/[0-9]*/fdinfo/ r,
+@{PROC}/[0-9]*/map_files/ r,
+@{PROC}/[0-9]*/ns/ r,
+
+@{PROC}/sys/kernel/panic w,
```

Comment:
This allows causing a kernel panic, why do we want to have this permission?

Reply:
It means this: "The value in this file represents the number of seconds the kernel waits before rebooting on a panic. When you use the software watchdog, the recommended setting is 60."

The reason is here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/container_manager_linux.go#L346

Reply:
I added a comment for this.